### DISCUSSION TITLE
Using preview versions in WAU to be installed on production clients

### DISCUSSION THREAD

**Discussion by KnifMelti** [Romanitho/Winget-AutoUpdate#61] [General]
@Romanitho 
When Microsoft releases the next stable version I guess it's gonna be rolled out via the Store backend?
Will clients with this preview version installed get the upgrade too or are you planning to build some sort of auto upgrade of Windows Package Manager into WAU?

**Comment by Romanitho** [comment]
I made few tests installing older versions of Winget from their Github repo. If a version (even preview ones) is older than the current MS Store one, it gets updated. So nothing to do for us with WAU :)
But I'll maintain the latest "stable" release in the script once the current preview one will be outdated. Hope I'm clear :)

**Reply by Romanitho** [reply]
So our computers get updated with latest Winget automatically from the store. It seems that we cannot have the stable version from their GitHub as they publish it through the store.

**Reply by KnifMelti** [reply]
Yes, and still Windows Package Manager v1.2.11601 (which is 1.17.11601.0) is the only existing in Store (from Sweden).
They've now started the 1.4 development builds...

**Reply by Romanitho** [reply]
in France too

**Reply by KnifMelti** [reply]
Aah...
https://github.com/microsoft/winget-cli/releases
Windows Package Manager **1.3.2091** Pre-release
_This release represents our **fourth** Windows Package Manager 1.3 release candidate build..._

**Reply by Romanitho** [reply]
Thanks :)

**Reply by KnifMelti** [reply]
Downgraded  and ran an update in Microsoft Store local on my computer from Sweden
...it automatically upgraded to **1.18.2091.0 (1.3.2091)**!!!!!!!
So, official now!

![image](https://user-images.githubusercontent.com/102996177/184466717-75bbb7f3-104c-4df8-bcfe-3acb2897dbc7.png)

**Reply by KnifMelti** [reply]
But now when I as a user runs winget --version:
`v1.3.2091`

When WAU runs in System contrext:
`13:22:47 - Winget Version: v1.3.0-preview`

So, the Get-WingetCmd.ps1 doesn't detect the latest version:
```
#Get WinGet Path (if admin context)
$ResolveWingetPath = Resolve-Path "C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_*_x64__8wekyb3d8bbwe"
if ($ResolveWingetPath) {
    #If multiple version, pick last one
    $WingetPath = $ResolveWingetPath[-1].Path
}
```
I have two versions now:
```
2022-08-13  05:19    <DIR>          Microsoft.DesktopAppInstaller_1.18.2091.0_neutral_split.language-sv_8wekyb3d8bbwe
2022-08-13  05:19    <DIR>          Microsoft.DesktopAppInstaller_1.18.2091.0_neutral_split.scale-100_8wekyb3d8bbwe
2022-08-13  05:19    <DIR>          Microsoft.DesktopAppInstaller_1.18.2091.0_x64__8wekyb3d8bbwe
2022-06-09  20:36    <DIR>          Microsoft.DesktopAppInstaller_1.18.431.0_x64__8wekyb3d8bbwe
2022-08-13  05:19    <DIR>          Microsoft.DesktopAppInstaller_2022.728.1939.0_neutral_~_8wekyb3d8bbwe
```

**Reply by Romanitho** [reply]
oh yes... it might be due to version order :/ (and sorry, i'm on holidays till end of august, so might be slower to reply ^^)

**Reply by Romanitho** [reply]
I guess the script picks the last line in the available "resolve-path". So it takes the last from these two lines:
```
2022-08-13  05:19    <DIR>          Microsoft.DesktopAppInstaller_1.18.2091.0_x64__8wekyb3d8bbwe
2022-06-09  20:36    <DIR>          Microsoft.DesktopAppInstaller_1.18.431.0_x64__8wekyb3d8bbwe
```
So we need to find a way to sort the result in function of the Winget version.

**Reply by KnifMelti** [reply]
See my Pull

**Reply by Romanitho** [reply]
I saw. I adapted your pull with a regex comparaison between version number inside the folder name. I think we can go further with getting the version of each winget.exe and pick the good one.

**Reply by KnifMelti** [reply]
SDont eforget **Invoke-PostUpdateActions**...

**Reply by Romanitho** [reply]
?

**Reply by KnifMelti** [reply]
The check exists in that function too

**Reply by Romanitho** [reply]
oh, true. yes
